pkg/controller/sync/version/manager.go

```diff
@@ -175,8 +175,12 @@ func (m *VersionManager) Update(resource VersionedResource,
 	}
 
 	if oldStatus != nil && util.PropagatedVersionStatusEquivalent(oldStatus, status) {
+		m.Unlock()
```

Comment:
Can we use a defer for this unlock and the one on line 191 or does `writeVersion` expect the version manager to be unlocked?

Reply:
Holding a lock while making network calls is risky, which is why `writeVersion` is called after unlock.

Reply:
Then what about lock & use & unlock? Scattering unlock for returns seems not good for maintainance.

Reply:
2 unlocks - one before exit and one before writing changes - doesn't strike me as 'scattering'.  How is that worse than adding lock/unlock pairs everywhere?

Reply:
Agreed, let's not hold a lock across network calls.

Reply:
I've added a comment indicating why the lock needs to be released before `writeVersion` is called.